### ISSUE TITLE
Debug flag for the dispatcher

### ISSUE THREAD

**Issue by ferrannp** [kenwheeler/mcfly#27]
As it is written here:

http://www.code-experience.com/async-requests-with-react-js-and-flux-revisited/
*There should be only one channel for all state changes: The Dispatcher. This makes debugging easy because it just requires a single console.log in the dispatcher to observe every single state change trigger.*

I am missing this "console.log" when the Dispatcher is using "dispatch" so it's very easy and fast to see every time that some action has been dispatched (and I don't need console.log in all the Stores).

Will you be able to add some debug flag to set this console.log on or off? Something like:

```
Dispatcher.dispatch(payload)
if(DEBUG){
  console.log(payload)
}
```

Or maybe like React does:
```
if ("production" !== process.env.NODE_ENV){
  console.log(payload)
}
```
Thanks in advance!

**Comment by balanceiskey** [comment]
An alternate you might consider trying that we've had success with is simply creating a DebugStore whose only responsibility is to log payloads.

**Comment by ferrannp** [comment]
@balanceiskey indeed, that could be a good approach... However, I think adding the second suggested piece of code would be better.

**Comment by buberdds** [comment]
@ferrannp Yeah, this piece of code won't hurt anyone and is quite useful +1

**Comment by kenwheeler** [comment]
@balanceiskey @ferrannp I'll be reviewing this this weekend.

**Comment by kenwheeler** [comment]
I'm not sure if implicitly including a debug mode, like example 2 is really desired as a core feature. That said, does anyone have any ideas around passing a debug flag? I'm having difficulty creating a global debug flag with browserify.

**Comment by tomatau** [comment]
Perhaps attach a debug flag to the McFly instance?

Webpack has good functionality for global flags but a modular framework should definitely not assume a build process, also, it's nice to support people who like to add their scripts individually to an html file!

I find the debug logging store to be the most flexible solution here!

**Comment by kraf** [comment]
I've been just dropping this snippet in my main entry points:
```
var Flux = require('wherever-you-build-flux');
Flux.dispatcher.register(function debugOutput(payload) {
	if(localStorage.debugDispatch === "true") {
		console.log(Date(), payload.actionType, payload);
	}
});
```
I like being able to enable/disable this in the console while the application is running.

@kenwheeler: Browserify gives you access to `process.env`. I'm pretty sure you can read anything from the environment, not just NODE_ENV.
